src/expr/src/scalar/func/impls/string.rs

```diff
@@ -688,6 +688,10 @@ impl<'a> EagerUnaryFunc<'a> for CastStringToVarChar {
         .nullable(input.nullable)
     }
 
+    fn could_error(&self) -> bool {
+        !self.fail_on_len || self.length.is_none()
+    }
+
```

Comment:
Would it be valid to add the same to `CastStringToChar`?

Reply:
I haven't looked closely, but almost certainly. I can try to make a pass on the unary funcs today.